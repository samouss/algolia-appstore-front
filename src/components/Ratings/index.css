.Ratings {
  color: #ff9501;
}
