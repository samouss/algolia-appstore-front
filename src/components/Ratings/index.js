/* eslint-disable react/no-array-index-key */

import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Star = () => (
  <i className="fa fa-star" aria-hidden="true" />
);

const StarEmpty = () => (
  <i className="fa fa-star-o" aria-hidden="true" />
);

const StarHalf = () => (
  <i className="fa fa-star-half-o" aria-hidden="true" />
);

const Ratings = ({ note, on }) => (
  <div styleName="Ratings">
    {new Array(on).fill(null).map((_, index) => {
      const next = index + 1;

      return (index < note && note < next) ? <StarHalf key={index} />
      : (next <= note) ? <Star key={index} />
      : <StarEmpty key={index} />;
    })}
  </div>
);

Ratings.propTypes = {
  note: PropTypes.number.isRequired,
  on: PropTypes.number.isRequired,
};

export default Ratings;
