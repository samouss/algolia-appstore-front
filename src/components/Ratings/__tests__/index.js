import React from 'react';
import { shallow } from 'enzyme';
import Ratings from '../index';

describe('<Ratings />', () => {
  const defaultProps = {
    note: 0,
    on: 5,
  };

  it('expect to render note 0', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Ratings
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render note 3', () => {
    const props = {
      ...defaultProps,
      note: 3,
    };

    const component = shallow(
      <Ratings
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render note 3.5', () => {
    const props = {
      ...defaultProps,
      note: 3.5,
    };

    const component = shallow(
      <Ratings
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render note 5', () => {
    const props = {
      ...defaultProps,
      note: 5,
    };

    const component = shallow(
      <Ratings
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render note 5 on 10', () => {
    const props = {
      ...defaultProps,
      note: 5,
      on: 10,
    };

    const component = shallow(
      <Ratings
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
