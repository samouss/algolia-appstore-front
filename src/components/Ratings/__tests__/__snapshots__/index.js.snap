// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ratings /> expect to render note 0 1`] = `
<div
  style={
    Object {
      "color": "#ff9501",
    }
  }
>
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
</div>
`;

exports[`<Ratings /> expect to render note 3 1`] = `
<div
  style={
    Object {
      "color": "#ff9501",
    }
  }
>
  <Star />
  <Star />
  <Star />
  <StarEmpty />
  <StarEmpty />
</div>
`;

exports[`<Ratings /> expect to render note 3.5 1`] = `
<div
  style={
    Object {
      "color": "#ff9501",
    }
  }
>
  <Star />
  <Star />
  <Star />
  <StarHalf />
  <StarEmpty />
</div>
`;

exports[`<Ratings /> expect to render note 5 1`] = `
<div
  style={
    Object {
      "color": "#ff9501",
    }
  }
>
  <Star />
  <Star />
  <Star />
  <Star />
  <Star />
</div>
`;

exports[`<Ratings /> expect to render note 5 on 10 1`] = `
<div
  style={
    Object {
      "color": "#ff9501",
    }
  }
>
  <Star />
  <Star />
  <Star />
  <Star />
  <Star />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
</div>
`;

exports[`<Ratings /> expect to render with custom color 1`] = `
<div
  style={
    Object {
      "color": "red",
    }
  }
>
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
  <StarEmpty />
</div>
`;
